android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/ListItemProxy.java

```diff
@@ -53,6 +54,11 @@ private void fireItemClick(String event, Object data)
 		if (event.equals(TiC.EVENT_CLICK) && data instanceof HashMap) {
 			KrollDict eventData = new KrollDict((HashMap) data);
 			TiViewProxy source = (TiViewProxy) eventData.get(TiC.EVENT_PROPERTY_SOURCE);
+
+			// FIXME: We should not need to create a placeholder proxy for each item. ListView needs refactoring to remedy this.
+			source = (TiViewProxy) KrollProxy.createProxy(source.getClass(), source.getKrollObject(), new Object[0],
+														  source.getCreationUrl().url);
+			eventData.put(TiC.EVENT_PROPERTY_SOURCE, source);
```

Comment:
We're not checking if `source` is `null` before calling `source.getClass()` and its other methods. Should we move this code into the if-block below? (Will the `!source.equals(this)` below be an issue?)

Also, the `createProxy()` method will call the given `KrollObject`'s `setProxySupport()` method, which replaces the previous proxy object's listeners. But this "might" be okay in this case since I don't think you set "click" listeners on it anyways.
https://github.com/appcelerator/titanium_mobile/blob/master/android/runtime/common/src/java/org/appcelerator/kroll/KrollObject.java#L42